### PR TITLE
executor: make seek value with the same length of the index columns.

### PR DESCRIPTION
**Pull request by coocood**
For composite index with 2 columns, if we just pass one value in `Seek`, we will get the wrong result.
So this change create the seek value with the length of the composite index.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: kv/index_iter_test.go [hunk 1]
```diff
@@ -143,3 +143,23 @@ func (s *testIndexSuite) TestIndex(c *C) {
...
+	txn, err := s.s.Begin()
+	c.Assert(err, IsNil)
+
+	values := []interface{}{"abc", "def"}
+	err = index.Create(txn, values, 1)
+	c.Assert(err, IsNil)
+
+	index2 := kv.NewKVIndex("i", "test", 1, false)
+	iter, hit, err := index2.Seek(txn, []interface{}{"abc", nil})
+	defer iter.Close()
```
Comment (qiuyesuifeng) [inline kv/index_iter_test.go]: If `iter == nil`, this will be panic?

File: executor/executor.go [hunk 2]
```diff
@@ -177,8 +178,8 @@ func (e *IndexRangeExec) Fields() []*ast.ResultField {
 // Next implements Executor Next interface.
 func (e *IndexRangeExec) Next() (*Row, error) {
 	if e.iter == nil {
-		seekVals := make([]interface{}, len(e.lowVals))
-		for i := 0; i < len(seekVals); i++ {
+		seekVals := make([]interface{}, len(e.scan.idx.Columns))
```
Comment (siddontang) [inline executor/executor.go]: len(e.scan.idx.Columns) is not equal to len(e.lowVals) ?

Comment (coocood) [inline executor/executor.go]: Yes, `e.lowVals` may be shorter.

File: kv/index_iter_test.go (hunk 1 above)
Comment (coocood) [inline kv/index_iter_test.go]: It's in test case, panic means test failed.

Comment (siddontang) [inline kv/index_iter_test.go]: don't panic, check err first.

Comment (qiuyesuifeng) [inline kv/index_iter_test.go]: I think it is better to check err first. Panic is not a good thing even if it is in test.

**Comment by siddontang** [comment]
LGTM

**Pull request merged**